internal/operations/operation_updater.go

```diff
@@ -207,26 +206,20 @@ func (ou *operationUpdater) doBatchUpdate(ctx context.Context, updates []*core.O
 	if len(opIDs) == 0 {
 		return nil
 	}
-	// TODO: cache these operation queries
-	opFilter := database.OperationQueryFactory.NewFilter(ctx).In("id", opIDs)
-	ops, _, err := ou.database.GetOperations(ctx, ou.manager.namespace, opFilter)
+	ops, err := ou.manager.getOperationsCached(ctx, opIDs)
 	if err != nil {
 		return err
 	}
 
 	// Get all the transactions for these operations
-	txIDs := make([]driver.Value, 0, len(ops))
+	var transactions []*core.Transaction
 	for _, op := range ops {
 		if op.Transaction != nil {
-			txIDs = append(txIDs, op.Transaction)
-		}
-	}
-	var transactions []*core.Transaction
-	if len(txIDs) > 0 {
-		txFilter := database.TransactionQueryFactory.NewFilter(ctx).In("id", txIDs)
-		transactions, _, err = ou.database.GetTransactions(ctx, ou.manager.namespace, txFilter)
-		if err != nil {
-			return err
+			transaction, err := ou.txHelper.GetTransactionByIDCached(ctx, op.Transaction)
```

Comment:
This could also help reduce the load, had forgotten we did this extra indirected lookup, just to be able to attache any associated Blockchain TXID to the FF TX 👍

Reply:
The line that made me think a bit is below (outside the comment-able section):

```go
		if err := ou.txHelper.AddBlockchainTX(ctx, tx, update.BlockchainTXID); err != nil {
```

The most important thing from a cache freshness perspective, is that this operation ends up as additive for any blockchain transaction ID. Right now we only have zero/1 for all our ops, so very unlikely to be able to test this. But I will try and dig into that function a bit if I don't naturally come to it in this review, to see if there's any cache issue inside it.

Reply:
Ok, so `AddBlockchainTX` just trusts the TX object that comes in is up to date - and adds to its list.

It does call `updateTransactionsCache` within that function - so I believe we're all good here.